agent/task.go

```diff
@@ -148,6 +148,11 @@ func (a *Agent) runPreTaskCommands(ctx context.Context, tc *taskContext) error {
 		}
 		if taskGroup.SetupGroup != nil {
 			opts.shouldSetupFail = taskGroup.SetupGroupFailTask
+			if taskGroup.SetupGroupTimeoutSecs > 0 {
+				fmt.Println("THIS SHOULD BE PRETTY SHORT")
+				ctx, cancel = context.WithTimeout(ctx, time.Duration(taskGroup.SetupGroupTimeoutSecs)*time.Second)
```

Comment:
This works, but I think it's a bit careless with the context (the code that was already here did it wrong too). I would use a new variable for the context that you wrapped to make it obvious it's not the same as the one passed in, and also do an if...else with the a.withCallbackTimeout so that you're not wrapping the context twice